quinn-proto/src/connection/datagrams.rs

```diff
@@ -149,6 +149,27 @@ impl DatagramState {
         Ok(was_empty)
     }
 
+    /// Discard outgoing datagrams with a payload larger than `max_payload` bytes
+    ///
+    /// Used to ensure that reductions in MTU don't get us stuck in a state where we have a datagram
+    /// queued but can't send it.
+    pub(super) fn drop_oversized(&mut self, max_payload: usize) {
+        for i in (0..self.outgoing.len()).rev() {
```

Comment:
Couldn't this be a call to `self.outgoing.retain()`?

Reply:
Good catch.